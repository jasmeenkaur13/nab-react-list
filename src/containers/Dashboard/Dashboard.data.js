export const ListsOfData = [{
    id: 1,
    Name: "Jasmeen",
    Age: 32,
    RequestedAccess: "jira",
    RequestedAccessName: "JIRA",
    ApprovedBy: "Jatin",
    markCompleted: false,
},
{
    id: 2,
    Name: "Hridaye",
    Age: 32,
    RequestedAccess: "confluence",
    RequestedAccessName: "Confluence",
    ApprovedBy: "Jatin",
    markCompleted: true,
},
{
    id: 3,
    Name: "Hanaaya",
    Age: 32,
    RequestedAccess: "payadvice",
    RequestedAccessName: "Pay Advice",
    ApprovedBy: "Mark",
    markCompleted: false,
},
{
    id: 4,
    Name: "Scott",
    Age: 34,
    RequestedAccess: "jira",
    RequestedAccessName: "JIRA",
    ApprovedBy: "Mark",
    markCompleted: false,
}];