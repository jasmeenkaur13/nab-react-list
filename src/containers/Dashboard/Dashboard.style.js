import styled from 'styled-components';
import Form from "react-bootstrap/Form";
import Nav from "react-bootstrap/Nav";

export const StyledUnorderedList = styled.ul`
list-style-type: none;
`;

export const StyledListContainer = styled.div`
display:flex;
`;

export const StyledListContainerHeading = styled.div`
display:flex;
margin-left:2rem;
font-weight: bold;
margin-top:3rem;
`;

export const StyledImage = styled.img`
height: 3rem;
width: 7rem;
`;

export const StyledNavigationLink = styled(Nav.Link)`
color:gray;
&:hover,
&:focus{
  color:gray;
}

`;

export const StyledLabel = styled(Form.Label)`
padding-top: .5rem;
line-height: normal;
width:10rem;
height:3rem;
border-style:solid;
text-align:center;
font-size: 15px;
border-right: ${props => props.isLast === 'true' ? 'solid' : 'none'}
`;

export const StyledCheckBox = styled(Form.Check)`
width:10rem;
font-size: 15px;
height:3rem;
border-style:solid;
padding-left: 2.5rem;
padding-top: .5rem;
.form-check-input{
  border: 1px solid lightgray;
}
.form-check-input:checked {
  background-color: gray;
  border-color: gray;
}
.form-check-input:focus,
.form-check-input:focus-visible {
  box-shadow: none;
}
`;
