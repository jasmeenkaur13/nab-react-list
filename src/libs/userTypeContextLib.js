import { useContext, createContext } from "react";

export const UserContext = createContext('readonly');

export function useUserContext() {
  return useContext(UserContext);
}