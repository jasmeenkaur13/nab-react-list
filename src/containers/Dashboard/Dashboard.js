import React, { useState } from 'react';
import Navbar from "react-bootstrap/Navbar";
import NabIcon from "../../images/logo.png"
import { useAppContext } from "../../libs/contextLib";
import { useUserContext } from "../../libs/userTypeContextLib";
import { useHistory } from "react-router-dom";
import {
    StyledUnorderedList, StyledListContainer, StyledLabel,
    StyledCheckBox, StyledListContainerHeading, StyledNavigationLink,
    StyledImage
} from "./Dashboard.style"

const ListsOfData = [{
    id: 1,
    Name: "Jasmeen",
    Age: 32,
    RequestedAccess: "jira",
    RequestedAccessName: "JIRA",
    ApprovedBy: "Jatin",
    markCompleted: false,
},
{
    id: 2,
    Name: "Hridaye",
    Age: 32,
    RequestedAccess: "confluence",
    RequestedAccessName: "Confluence",
    ApprovedBy: "Jatin",
    markCompleted: true,
},
{
    id: 3,
    Name: "Hanaaya",
    Age: 32,
    RequestedAccess: "payadvice",
    RequestedAccessName: "Pay Advice",
    ApprovedBy: "Mark",
    markCompleted: false,
},
{
    id: 4,
    Name: "Scott",
    Age: 34,
    RequestedAccess: "jira",
    RequestedAccessName: "JIRA",
    ApprovedBy: "Mark",
    markCompleted: false,
}];

export default function Dashboard() {
    const { userHasAuthenticated } = useAppContext();
    const { userType } = useUserContext();
    const [list, setList] = useState(ListsOfData);


    const history = useHistory();

    const handleChangeCheckbox = id => {
        setList(
            list.map(item => {
                if (item.id === id) {
                    return { ...item, markCompleted: !item.markCompleted };
                } else {
                    return item;
                }
            })
        );
    };

    function handleLogout() {
        userHasAuthenticated(false);
        history.push("/");

    }

    const listItems = (
        list.map((item, index) => {
            if (userType === 'admin' || userType === item.RequestedAccess) {
                return (
                    <li key={'list-' + index}>
                        <StyledListContainer key={'list-' + index}>
                            <StyledLabel>{item.Name}</StyledLabel>
                            <StyledLabel>{item.Age}</StyledLabel>
                            <StyledLabel>{item.ApprovedBy}</StyledLabel>
                            <StyledLabel>{item.RequestedAccessName}</StyledLabel>
                            <StyledCheckBox label={item.markCompleted ? "Completed" : "Incomplete"}
                                checked={item.markCompleted}
                                disabled={item.markCompleted}
                                onChange={() => handleChangeCheckbox(item.id)} />
                        </StyledListContainer>
                    </li>)
            }
            if (userType === 'readonly') {
                return (
                    <li key={'list-' + index}>
                        <StyledListContainer key={'list-' + index}>
                            <StyledLabel>{item.Name}</StyledLabel>
                            <StyledLabel>{item.Age}</StyledLabel>
                            <StyledLabel>{item.ApprovedBy}</StyledLabel>
                            <StyledLabel>{item.RequestedAccessName}</StyledLabel>
                            <StyledLabel isLast='true'>{item.markCompleted ? "Completed" : "Incomplete"}</StyledLabel>
                        </StyledListContainer>
                    </li>)
            }
        }));
    return (
        <div>
            <Navbar collapseOnSelect bg="dark" expand="md" className="mb-3">
                <Navbar.Brand className="font-weight-bold text-muted">
                    <StyledImage src={NabIcon} />
                </Navbar.Brand>
                <Navbar.Toggle />
                <Navbar.Collapse className="justify-content-end">
                    <StyledNavigationLink onClick={handleLogout}>Logout</StyledNavigationLink>
                </Navbar.Collapse>
            </Navbar>
            <h2>Dashboard</h2>
            <StyledListContainerHeading key={'list-heading'}>
                <StyledLabel>Name</StyledLabel>
                <StyledLabel>Age</StyledLabel>
                <StyledLabel>Approved By</StyledLabel>
                <StyledLabel>Requested Access</StyledLabel>
                <StyledLabel isLast='true'>Completed</StyledLabel>
            </StyledListContainerHeading>
            <StyledUnorderedList id='data'>{listItems}</StyledUnorderedList>
        </div>
    );
}