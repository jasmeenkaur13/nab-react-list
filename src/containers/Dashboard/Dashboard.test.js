import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from "./Dashboard";
import { UserContext } from "../../libs/userTypeContextLib";
import { AppContext } from "../../libs/contextLib";


it('renders correctly when there is a no data for invalid usertype', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'hi' }}>
        <Dashboard />
      </UserContext.Provider>)
    </AppContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a data for readonly usertype with no update permissions', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'readonly' }}>
        <Dashboard />
      </UserContext.Provider>)
    </AppContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a data for jira usertype with update permissions', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'jira' }}>
        <Dashboard />
      </UserContext.Provider>)
    </AppContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a data for admin usertype with update permissions', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'admin' }}>
        <Dashboard />
      </UserContext.Provider>
    </AppContext.Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there is a data for admin usertype with updating the value', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'admin' }}>
        <Dashboard />
      </UserContext.Provider>
    </AppContext.Provider>);
  const checkbox = tree.root.findAllByType('input');
  renderer.act(checkbox[0].props.onChange);
  expect(tree).toMatchSnapshot();
});

it('renders correctly when we logout', () => {
  const accesstype = "readonly";
  const addItem = jest.fn()
  const tree = renderer.create(
    <AppContext.Provider value={{ addItem }}>
      <UserContext.Provider value={{ userType: 'admin' }}>
        <Dashboard />
      </UserContext.Provider>
    </AppContext.Provider>);
  const link = tree.root.findAllByType('a');
  renderer.act(() => { link[0].props.onClick });
  expect(tree).toMatchSnapshot();

});

