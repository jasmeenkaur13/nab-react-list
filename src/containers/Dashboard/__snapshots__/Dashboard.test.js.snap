// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is a data for admin usertype with update permissions 1`] = `
<div>
  <nav
    className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
  >
    <span
      className="font-weight-bold text-muted navbar-brand"
    >
      <img
        className="sc-hKFxyN hLMffp"
        src="logo.png"
      />
    </span>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <a
        className="sc-eCApnc bBUfuo nav-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Logout
      </a>
    </div>
  </nav>
  <h2>
    Dashboard
  </h2>
  <div
    className="sc-dlnjwi ckwOgj"
  >
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Name
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Age
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Approved By
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Requested Access
    </label>
    <label
      className="sc-jSFjdj bWZEOZ form-label"
      isLast="true"
    >
      Completed
    </label>
  </div>
  <ul
    className="sc-bdnxRM kUrNUp"
    id="data"
  >
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jasmeen
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hridaye
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Confluence
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Completed
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hanaaya
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Pay Advice
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Scott
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          34
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders correctly when there is a data for admin usertype with updating the value 1`] = `
<div>
  <nav
    className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
  >
    <span
      className="font-weight-bold text-muted navbar-brand"
    >
      <img
        className="sc-hKFxyN hLMffp"
        src="logo.png"
      />
    </span>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <a
        className="sc-eCApnc bBUfuo nav-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Logout
      </a>
    </div>
  </nav>
  <h2>
    Dashboard
  </h2>
  <div
    className="sc-dlnjwi ckwOgj"
  >
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Name
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Age
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Approved By
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Requested Access
    </label>
    <label
      className="sc-jSFjdj bWZEOZ form-label"
      isLast="true"
    >
      Completed
    </label>
  </div>
  <ul
    className="sc-bdnxRM kUrNUp"
    id="data"
  >
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jasmeen
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Completed
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hridaye
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Confluence
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Completed
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hanaaya
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Pay Advice
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Scott
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          34
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`renders correctly when there is a data for jira usertype with update permissions 1`] = `
Array [
  <div>
    <nav
      className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
    >
      <span
        className="font-weight-bold text-muted navbar-brand"
      >
        <img
          className="sc-hKFxyN hLMffp"
          src="logo.png"
        />
      </span>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="justify-content-end navbar-collapse collapse"
      >
        <a
          className="sc-eCApnc bBUfuo nav-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
      </div>
    </nav>
    <h2>
      Dashboard
    </h2>
    <div
      className="sc-dlnjwi ckwOgj"
    >
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Name
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Age
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Approved By
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Requested Access
      </label>
      <label
        className="sc-jSFjdj bWZEOZ form-label"
        isLast="true"
      >
        Completed
      </label>
    </div>
    <ul
      className="sc-bdnxRM kUrNUp"
      id="data"
    >
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Jasmeen
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            32
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Jatin
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            JIRA
          </label>
          <div
            className="sc-gKAaRy eXCtXX form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="form-check-label"
              title=""
            >
              Incomplete
            </label>
          </div>
        </div>
      </li>
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Scott
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            34
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Mark
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            JIRA
          </label>
          <div
            className="sc-gKAaRy eXCtXX form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              disabled={false}
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="form-check-label"
              title=""
            >
              Incomplete
            </label>
          </div>
        </div>
      </li>
    </ul>
  </div>,
  ")",
]
`;

exports[`renders correctly when there is a data for readonly usertype with no update permissions 1`] = `
Array [
  <div>
    <nav
      className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
    >
      <span
        className="font-weight-bold text-muted navbar-brand"
      >
        <img
          className="sc-hKFxyN hLMffp"
          src="logo.png"
        />
      </span>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="justify-content-end navbar-collapse collapse"
      >
        <a
          className="sc-eCApnc bBUfuo nav-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
      </div>
    </nav>
    <h2>
      Dashboard
    </h2>
    <div
      className="sc-dlnjwi ckwOgj"
    >
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Name
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Age
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Approved By
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Requested Access
      </label>
      <label
        className="sc-jSFjdj bWZEOZ form-label"
        isLast="true"
      >
        Completed
      </label>
    </div>
    <ul
      className="sc-bdnxRM kUrNUp"
      id="data"
    >
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Jasmeen
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            32
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Jatin
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            JIRA
          </label>
          <label
            className="sc-jSFjdj bWZEOZ form-label"
            isLast="true"
          >
            Incomplete
          </label>
        </div>
      </li>
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Hridaye
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            32
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Jatin
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Confluence
          </label>
          <label
            className="sc-jSFjdj bWZEOZ form-label"
            isLast="true"
          >
            Completed
          </label>
        </div>
      </li>
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Hanaaya
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            32
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Mark
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Pay Advice
          </label>
          <label
            className="sc-jSFjdj bWZEOZ form-label"
            isLast="true"
          >
            Incomplete
          </label>
        </div>
      </li>
      <li>
        <div
          className="sc-gtsrHT eHuwNW"
        >
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Scott
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            34
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            Mark
          </label>
          <label
            className="sc-jSFjdj hzwScC form-label"
          >
            JIRA
          </label>
          <label
            className="sc-jSFjdj bWZEOZ form-label"
            isLast="true"
          >
            Incomplete
          </label>
        </div>
      </li>
    </ul>
  </div>,
  ")",
]
`;

exports[`renders correctly when there is a no data for invalid usertype 1`] = `
Array [
  <div>
    <nav
      className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
    >
      <span
        className="font-weight-bold text-muted navbar-brand"
      >
        <img
          className="sc-hKFxyN hLMffp"
          src="logo.png"
        />
      </span>
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="justify-content-end navbar-collapse collapse"
      >
        <a
          className="sc-eCApnc bBUfuo nav-link"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="button"
        >
          Logout
        </a>
      </div>
    </nav>
    <h2>
      Dashboard
    </h2>
    <div
      className="sc-dlnjwi ckwOgj"
    >
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Name
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Age
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Approved By
      </label>
      <label
        className="sc-jSFjdj hzwScC form-label"
      >
        Requested Access
      </label>
      <label
        className="sc-jSFjdj bWZEOZ form-label"
        isLast="true"
      >
        Completed
      </label>
    </div>
    <ul
      className="sc-bdnxRM kUrNUp"
      id="data"
    />
  </div>,
  ")",
]
`;

exports[`renders correctly when we logout 1`] = `
<div>
  <nav
    className="mb-3 navbar navbar-expand-md navbar-light bg-dark"
  >
    <span
      className="font-weight-bold text-muted navbar-brand"
    >
      <img
        className="sc-hKFxyN hLMffp"
        src="logo.png"
      />
    </span>
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <a
        className="sc-eCApnc bBUfuo nav-link"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
      >
        Logout
      </a>
    </div>
  </nav>
  <h2>
    Dashboard
  </h2>
  <div
    className="sc-dlnjwi ckwOgj"
  >
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Name
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Age
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Approved By
    </label>
    <label
      className="sc-jSFjdj hzwScC form-label"
    >
      Requested Access
    </label>
    <label
      className="sc-jSFjdj bWZEOZ form-label"
      isLast="true"
    >
      Completed
    </label>
  </div>
  <ul
    className="sc-bdnxRM kUrNUp"
    id="data"
  >
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jasmeen
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hridaye
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Jatin
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Confluence
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={true}
            className="form-check-input"
            disabled={true}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Completed
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Hanaaya
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          32
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Pay Advice
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
    <li>
      <div
        className="sc-gtsrHT eHuwNW"
      >
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Scott
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          34
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          Mark
        </label>
        <label
          className="sc-jSFjdj hzwScC form-label"
        >
          JIRA
        </label>
        <div
          className="sc-gKAaRy eXCtXX form-check"
        >
          <input
            checked={false}
            className="form-check-input"
            disabled={false}
            onChange={[Function]}
            type="checkbox"
          />
          <label
            className="form-check-label"
            title=""
          >
            Incomplete
          </label>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
